### DISCUSSION TITLE
Setting upstream proxy on requests intercepted using a module.

### DISCUSSION THREAD

**Discussion by Ghxst** [mitmproxy/mitmproxy#6354] [Q&A] [answered]
Hey I was wondering if / how I can set an upstream proxy when I intercept a request inside a module.
During my research I found the following code linked in a comment but I am not able to find this example in the current examples directory and I am not sure if this works when mitmproxy is started in default mode.
https://github.com/mitmproxy/mitmproxy/blob/v5.2/examples/contrib/change_upstream_proxy.py
And there's an SO answer here that seems relevant, but unsure if this is the correct way of doing this.
https://stackoverflow.com/a/72094131/12177892

Any pointers are appreciated.

**Comment by Prinzhorn** [comment, marked as answer]
> but I am not able to find this example in the current examples directory

Hm, it's right there? https://github.com/mitmproxy/mitmproxy/blob/main/examples/contrib/change_upstream_proxy.py
